### PR TITLE
Setup Hugo site with Docsy Theme

### LINKED ISSUE #5: Setup a Hugo website with a Docsy theme
* [Hugo](https://gohugo.io/)
* [Docsy theme](https://www.docsy.dev/)

### PR DESCRIPTION
**Pull request by nate-double-u**
Closes #5, #10

First pass at notaryproject.dev - cloning etcd-io/website.
Deploy preview: https://deploy-preview-9--notarydev.netlify.app/

### CODE REVIEW

**Review by chalin — COMMENTED** [review]
Hmm, maybe you didn't start from the latest version of etcd.io? For example, this PR contains `check-hugo.sh`.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+DOCKER_IMG = klakegg/hugo:ext-alpine
+DRAFT_ARGS = --buildDrafts --buildFuture  --buildExpired
+
+# Keep these consistent with the values in config.yaml and layouts/index.redirects:
+NEXTv=v3.5
+LATESTv=v3.5
```
Comment (chalin) [inline Makefile:6]: If you change both of these to v2, then the link checking should work.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[submodule "themes/docsy"]
+	path = themes/docsy
+	url = https://github.com/cncf/docsy.git
+	branch = notaryproject.dev
```
Comment (chalin) [inline .gitmodules:4]: 👍 for this new branch, and having based it on the most recent version of docsy.

File: check-hugo.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# based on gnossen's check_hugo.sh file:
+# https://github.com/grpc/grpc.io/blob/main/check_hugo.sh
+
+set -e
+
+command -v hugo >/dev/null || (echo "Hugo extended must be installed on your system." >/dev/stderr; exit 1)
+hugo version | grep -i extended >/dev/null || (echo "Your Hugo installation does not appear to be extended." >/dev/stderr; exit 1)
+
```
Comment (chalin) [inline check-hugo.sh:10]: We shouldn't be using this file.

File: config.yaml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  # This menu appears only if you have at least one [params.versions] set.
+  version_menu: Versions
+  versions:
+    latest: v2
+    all:
+      - v2
+      - v1
+    deprecation_warning: |-
+      the documentation is no longer actively maintained.
+        The page that you are viewing is the last archived version.
```
Comment (chalin) [inline config.yaml:91]: I wish that we had cleaned this up first. Maybe we can do the cleanup in this repo and back port a solution.

File: content/en/about.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+---
+title: About
+---
```
Comment (chalin) [inline content/en/about.md:3]: Are you sure that you want an about page? If not, drop this page.

File: content/en/docs/v2/_index.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+title: v2 docs
+cascade:
+  version: v2
+  versName: &name v2
+  git_version_tag: v2
+linkTitle: *name
+simple_list: true
+weight: -350 # Weight for doc version vX.Y should be -XY0
```
Comment (chalin) [inline content/en/docs/v2/_index.md:9]: Weight should be -200 if we're following the same convention.

File: content/en/docs/v1/_index.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+---
+title: v1 docs
+cascade:
+  version: &vers v1
+  git_version_tag: v1
+  is_deprecated: true
+linkTitle: *vers
+simple_list: true
+weight: -340
```
Comment (chalin) [inline content/en/docs/v1/_index.md:9]: Weight should be -100.

File: layouts/index.redirects [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{{ $latest := site.Params.versions.latest }}
+/docs/latest     /docs/{{ $latest }}/
+/docs/latest/*     /docs/{{ $latest }}/:splat
+
+/docs/current     /docs/next/
+/docs/current/*     /docs/next/:splat
+
```
Comment (chalin) [inline layouts/index.redirects:7]: We should eliminate current. That's an artifact of the way etcd.io was setup over time (IMHO), not really something we need to keep / setup here.

File: layouts/robots.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+Sitemap: {{ site.BaseURL }}/sitemap.xml
+User-agent: *
```
Comment (chalin) [inline layouts/robots.txt:2]: Drop this file.

File: package.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    "build": "hugo --cleanDestinationDir -e dev -DFE",
+    "check-links": "make clean-htmltest-dir check-internal-links",
+    "ci-check-links": "make clean-htmltest-dir get-link-checker check-internal-links",
+    "get-submodules": "git submodule update --init --recursive --depth 1",
+    "prebuild:preview": "./check-hugo.sh",
+    "prebuild:production": "./check-hugo.sh",
+    "prebuild": "./check-hugo.sh",
+    "preinstall": "npm run get-submodules",
+    "preserve": "./check-hugo.sh",
+    "serve": "netlify dev -c \"hugo serve --minify -DFE -w\""
```
Comment (chalin) [inline package.json:15]: Get the latest from etcd.io

**Review by chalin — CHANGES_REQUESTED** [review]
You'll need to get the latest etcd.io files. In particular, we don't want the version of `render-link.html` that is currently in this PR.

For details, see inlined comments.

File: layouts/_default/_markup/render-link.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    {{ with $url.Fragment }}{{ $fragment = printf "#%s" . }}{{ end -}}
+    {{ with .Page.GetPage $url.Path }}{{ $link = printf "%s%s" .RelPermalink $fragment }}{{ end -}}
+  {{ end -}}
+{{ end -}}
+<a href="{{ $link | safeURL }}"
+  {{- with .Title}} title="{{ . }}"{{ end -}}
+  {{- if $isRemote }} target="_blank" rel="noopener"{{ end -}}
+>
+{{- .Text | safeHTML -}}
+</a>
```
[the remark this answers is not in the document]
Comment (chalin) [reply layouts/_default/_markup/render-link.html:16]: Correction: get the version from https://github.com/etcd-io/website/pull/420.

**Comment by nate-double-u** [comment]
> Hmm, maybe you didn't start from the latest version of etcd.io? For example, this PR contains `check-hugo.sh`.

I think you're right -- I'm disappointed that I missed this. I'd been working between two computers and think I did it on one but had to redo some work on the other. Not an excuse, just an explanation of what I think happened. I'll try to do a diff and bring this repo up to speed.

**Comment by nate-double-u** [comment]
> ✔️ Deploy Preview for _notarydev_ ready!
> 
> 🔨 Explore the source changes: [57cc9b3](https://github.com/notaryproject/notaryproject.dev/commit/57cc9b37126086ea04fb462fa6446404c76de345)
> 
> 🔍 Inspect the deploy log: [https://app.netlify.com/sites/notarydev/deploys/61141de6b40a130009ae6af9](https://app.netlify.com/sites/notarydev/deploys/61141de6b40a130009ae6af9?utm_source=github&utm_campaign=bot_dl)
> 
> 😎 Browse the preview: [https://deploy-preview-9--notarydev.netlify.app](https://deploy-preview-9--notarydev.netlify.app?utm_source=github&utm_campaign=bot_dp)

This is new, I like that the Netlify bot now provides an info comment (with a deploy preview)

File: config.yaml (hunk 4 above)
Comment (nate-double-u) [reply config.yaml:91]: I opened a PR a little bit ago that looks at doing some of that cleanup: https://github.com/etcd-io/website/pull/354

File: content/en/docs/v2/_index.md (hunk 6 above)
Comment (nate-double-u) [reply content/en/docs/v2/_index.md:9]: Good catch, thx

File: layouts/index.redirects (hunk 8 above)
Comment (nate-double-u) [reply layouts/index.redirects:7]: I think you're right here. I'll remove it for this site.

File: layouts/shortcodes/metrics-list.html [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+      {{ if (findRE $regexp .Name) -}}
+        {{ $displayName := .Name | replaceRE "^notaryproject-metrics-" "" | replaceRE ".txt$" "" -}}
+        <td><a href="{{ .Name }}">{{ $displayName }}</a></td>
+      {{ end -}}
+    {{ if eq 5 (mod $counter 6) -}}
+    </tr>
+    {{ end -}}
+    {{ $counter = add $counter 1 -}}
+  {{ end -}}
+</table>
```
[the remark this answers is not in the document]
Comment (nate-double-u) [reply layouts/shortcodes/metrics-list.html:19]: Agreed, thanks

File: Makefile (hunk 1 above)
Comment (nate-double-u) [reply Makefile:6]: Changed, will reintroduce the link check in the next commit to test.

File: content/en/about.md (hunk 5 above)
Comment (nate-double-u) [reply content/en/about.md:3]: I think I'll keep it in for now, and have a discussion with the maintainers about it.

I've been looking at the issues/PRs list and can't find the discussion we had with the rationale behind cutting it. I remember agreeing, but it'd be good to link the issue. Do you recall where we had the conversation, @chalin?

For context, @SteveLasker, the whole site is basically _about_ the project, so a specific _about_ page can be a bit redundant (I know I've included an _about_ in the docs assessment IA, but included it as a placeholder for discussion--I should be more clear about that in the assessment itself).

**Comment by nate-double-u** [comment]
> Link checker failed... I'll disable it for now.

I've re-enabled it in the latest commit.

**Comment by nate-double-u** [comment]
@SteveLasker, in the footer I have the copyright line:
> © 2013-2021 Notary Project Authors

I doubt this is the correct date range, do you know which year we should use as the start date here?

File: content/en/about.md (hunk 5 above)
Comment (chalin) [reply content/en/about.md:3]: FYI: https://github.com/etcd-io/website/issues/434 :)

File: .editorconfig [hunk 13]
```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
```
Comment (chalin) [inline .editorconfig:5]: We don't want to force final nl in certain layout files, though it seems that VS Code ignores this ... ?

File: Makefile [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+HTMLTEST_DIR=tmp
+HTMLTEST?=htmltest # Specify as make arg if different
+# Use $(HTMLTEST) in PATH, if available; otherwise, we'll get a copy
+ifeq (, $(shell command -v $(HTMLTEST)))
+override HTMLTEST=$(HTMLTEST_DIR)/bin/htmltest
+endif
+
+docker-serve:
+	docker run --rm -it -v $(PWD):/src -p 1313:1313 $(DOCKER_IMG) server $(DRAFT_ARGS)
```
Comment (chalin) [inline Makefile:16]: Consider dropping docker support (I thought that that is what we had done elsewhere).

**Review by chalin — APPROVED** [review]
I'd drop the About page now (it is harder to remove things once they're introduced). Other than that, and the minor inlined comments, LGTM.

File: content/en/about.md (hunk 5 above)
Comment (nate-double-u) [reply content/en/about.md:3]: Thanks!

File: .editorconfig (hunk 13 above)
Comment (nate-double-u) [reply .editorconfig:5]: I'm happy to take it out -- we'll want to back-port the change. I'm not super happy with the way editor config plugins seem to only work some of the time across platforms/editors...

**Comment by nate-double-u** [comment]
Thanks for the feedback @chalin, I appreciate it!

**Pull request merged**